--- qtwebkit-opensource-src-5.3.2+dfsg/Source/WTF/wtf/Atomics.h~	2014-09-11 12:48:11.000000000 +0200
+++ qtwebkit-opensource-src-5.3.2+dfsg/Source/WTF/wtf/Atomics.h	2015-02-13 22:32:03.647863539 +0100
@@ -188,7 +188,7 @@
 #if CPU(X86_64)
     bool result;
     asm volatile(
-        "lock; cmpxchgq %3, %2\n\t"
+        "lock; cmpxchg %3, %2\n\t"
         "sete %1"
         : "+a"(expected), "=q"(result), "+m"(*location)
         : "r"(newValue)
